### ISSUE TITLE
no visible global function definition false positive when inside a function inside a script with the necessary library call

### ISSUE THREAD

**Issue by kendonB** [r-lib/lintr#1966]
```
library(tidyverse)
tmp <- function(){
    mtcars |>
        pull(mpg)
}
```

![image](https://github.com/r-lib/lintr/assets/5327505/239b65eb-18d0-4a11-9cd9-4fadf5ac06b3)

Could be a matter of opinion whether this is bad practice or not vs using `dplyr::pull`

**Comment by AshesITR** [comment]
Thanks for reporting.
This is a near-duplicate of #1147.

You can resolve the issue by loading `library(tidyverse)` in the R session before running `lint()`.

**Comment by USMortality** [comment]
According to the above screenshot, he is already loading `library(tidyverse)`, what's the issue?

**Comment by kendonB** [comment]
So this needs to be solved by the vscode implementation of lintr?

**Comment by USMortality** [comment]
I'm not sure, possibly...  I'm also having this issue, a possible workaround is to use e.g.
```
pull <- dplyr::pull
```

but, this even breaks when sourcing it from a second/common file, e.g.: https://github.com/r-lib/lintr/issues/2050

Other workarounds, I've found, don't work either:
```
## dplyr namespace: start
#' @importFrom dplyr pull
## dplyr namespace: end

utils::globalVariables(c("pull"))
```

**Comment by AshesITR** [comment]
What lintr could do to help would be to assume `library()` calls to be equivalent to what packages can do with `import(...)` in their namespace. This would fix this bug, but not anything related to code spanning multiple files.

What your IDE could help you with would be to source some predefined scripts (think some `helper.R`) _before_ linting. This would help with objects defined in said helper script but causes some overhead.

A last resort for false positives is to ignore them line-by-line using `# nolint: object_usage_linter.`, but this will not be viable if there are many such lints.

NB usually you'll actually want `source(..., local = TRUE)` for your code to play nicely with environments.

**Comment by MichaelChirico** [comment]
I _guess_ #2696 will solve this?

**Comment by AshesITR** [comment]
Using a `.lintr.R` config file and having that code load the necessary packages / helpers could also solve this issue.
